hoodie-client/src/test/java/com/uber/hoodie/common/HoodieMergeOnReadClientTestUtil.java

```diff
@@ -0,0 +1,339 @@
+                            + hoodieTable.getMetaClient().getBasePath());
+                }
+                //TODO(na) : find a better way to list partitions only
+                String partition = path.substring(hoodieTable.getMetaClient().getBasePath().length() + 1, path.length() - 2);
+                records.addAll(gerMergedRecords(fs, generateSplit(this.basePath, partition)));
+            }
+            return convertToDF(records);
+        } catch (Exception e) {
+            throw new HoodieException("Error reading hoodie dataset as a dataframe", e);
+        }
+    }
+
+    @Override
+    public Dataset<Row> readSince(String lastCommitTimestamp) {
+
+        List<HoodieInstant> commitsToReturn =
+                commitTimeline.findInstantsAfter(lastCommitTimestamp, Integer.MAX_VALUE)
+                        .getInstants().collect(Collectors.toList());
+
+        Schema schema = HoodieAvroUtils.addMetadataFields(Schema.parse(TRIP_EXAMPLE_SCHEMA));
+        List<GenericRecord> records = new ArrayList<GenericRecord>();
+        try {
+            // Go over the commit metadata, and obtain the new files that need to be read.
+            HashMap<String, String> fileIdToFullPath = new HashMap<>(); //NOTE : problem with fileId for parquet and log file is the same
+            List<String> fullPaths = new ArrayList<>();
+            for (HoodieInstant commit : commitsToReturn) {
+                HoodieCommitMetadata metadata =
+                        HoodieCommitMetadata.fromBytes(commitTimeline.getInstantDetails(commit).get());
+                // get files from each commit, and replace any previous versions
+                fullPaths.addAll(metadata.getFileIdAndFullPaths().values());
+                fileIdToFullPath.putAll(metadata.getFileIdAndFullPaths());
+            }
+
+            return convertToDF(getMergedRecords(fullPaths));
+        } catch (IOException e) {
+            throw new HoodieException("Error pulling data incrementally from commitTimestamp :" + lastCommitTimestamp, e);
+        }
+    }
+
+    @Override
+    public String latestCommit() { return commitTimeline.lastInstant().get().getTimestamp(); }
+
+    private HoodieWriteConfig getConfig(String basePath) {
+        return getConfigBuilder(basePath).build();
+    }
+
+    private HoodieWriteConfig.Builder getConfigBuilder(String basePath) {
+        return HoodieWriteConfig.newBuilder().withPath(basePath)
+                .withSchema(TRIP_EXAMPLE_SCHEMA).withParallelism(2, 2)
+                .withCompactionConfig(HoodieCompactionConfig.newBuilder().compactionSmallFileSize(1024 * 1024).build())
+                .withStorageConfig(HoodieStorageConfig.newBuilder().limitFileSize(1024 * 1024).build())
+                .forTable("test-trip-table").withIndexConfig(
+                        HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.BLOOM).build());
+    }
+
+    //generate splits for each <hoodiefile, List<logfile>> pairs
```

Comment:
Tried to use HoodieRealtimeInputFormat instead of this logic, the method public InputSplit[] getSplits(JobConf job, int numSplits) always gives back 0 splits.